src/pip/_internal/req/req_install.py

```diff
@@ -169,11 +170,15 @@ def from_req(cls, req, comes_from=None, isolated=False, wheel_cache=None):
             req = Requirement(req)
         except InvalidRequirement:
             raise InstallationError("Invalid requirement: '%s'" % req)
-        if req.url:
-            raise InstallationError(
-                "Direct url requirement (like %s) are not allowed for "
-                "dependencies" % req
-            )
+
+        if req.url and comes_from.link.netloc == PyPI.link_source:
+            # Explicitly blacklist pypi packages that depend on external urls
```

Comment:
This indentation seems weird.

Could you make both the comment and the statement the same indent, which is a multiple of 4?